### ISSUE TITLE
Uploading to keys containing @ fails with `error: 403: "The request signature we calculated does not match the signature you provided. Check your key and signing method."`

### ISSUE THREAD

**Issue by BogdanSorlea** [rlmcpherson/s3gof3r#112]
this fails:
```
gof3r cp testing_gzip\@bla.gz s3://ts-zfs-backups-eu-west-1/testing_gzip\@bla.gz --endpoint=s3-eu-west-1.amazonaws.com
gof3r error: 403: "The request signature we calculated does not match the signature you provided. Check your key and signing method."
```

this works (notice replacing @ with _ in the key name):
```
gof3r cp testing_gzip\@bla.gz s3://ts-zfs-backups-eu-west-1/testing_gzip_bla.gz --endpoint=s3-eu-west-1.amazonaws.com
```

the key name is mandatory (i.e. does not seem to be inferred from filename, so there is really no workaround for this problem)

AWS permits creating keys with @ in them (either through the web console or aws-cli tool)

**Comment by BogdanSorlea** [comment]
Same for colon `:` :|

**Comment by rlmcpherson** [comment]
Fixed by #113
